site/source/docs/api_reference/html5.h.rst

```diff
@@ -1698,6 +1698,12 @@ Functions
 	
 	.. note:: A gamepad does not show up as connected until a button on it is pressed.
 
+	.. note:: Gamepad API uses an array of gamepad state objects to return the state of each device. The devices are identified via the index they are present in in
+	this array. Because of that, if one first connects gamepad A, then gamepad B, and then disconnects gamepad A, the gamepad B shall not take the place of gamepad A,
+	so in this scenario, this function will still keep returning two for the count of connected gamepads, even though gamepad A is no longer present. To find the actual
+	number of connected gamepads, listen for the gamepadconnected and gamepaddisconnected events.
+	Consider the return value of this function as the largest value (-1) that can be passed to the function emscripten_get_gamepad_status().
+
```

Comment:
Actually, this PR has a subtle side effect of unifying certain unspecced behavioral difference we've seen in browsers (which is bug https://github.com/w3c/gamepad/issues/13):

If one first connects some gamepads and then disconnects them, Firefox will return an empty array `[]` in the list of connected gamepads, but Chrome will return an array which contains `[null, null, ..., null]` entries up to the number of gamepads that had been connected. Firefox and Chrome will because of that return a different value as the number of connected gamepads. (It's better to understand that number as the number of currently "potentially active" gamepad slots to sample)

This PR unifies this scenario, i.e. if one connects and then disconnects all gamepads, both Firefox and Chrome will then equally have `emscripten_get_num_gamepads()` return 0.